### DISCUSSION TITLE
Is Docker/Docker-compose a supported/maintained shaarli installation method?

### DISCUSSION THREAD

**Discussion by daks** [shaarli/Shaarli#2099] [Q&A] [answered]
I use Shaarli since several years and want to move my nginx/php-fpm installation to a Docker one, to simplify maintenance, upgrade and also improve security (I plan to use Docker rootless which I think is more secure).
Before starting to use the provided Docker or docker-compose files, I would like to know if it's a maintained method for Shaarli installation: I see that both files haven't been updated in 9 months/1 year (maybe for good reasons).

**Comment by nodiscc** [comment, marked as answer]
Hi,

the docker image is maintained ([this pull request](https://github.com/shaarli/Shaarli/pull/2088) further bumps the base alpine image version, but I didn't have time to test it yet).

The docker-compose file can be considered unmaintained (I don't use it, and other maintainers who do are currently inactive). I guess we should at least update the `traefik:1.7-alpine` image. Traefik 1.x images don't seem to be available anymore (https://hub.docker.com/_/traefik/) so it means upgrading to at least 2.x with potential breaking changes - hence some testing/adjustment would be needed. Someone [attempted it](https://github.com/shaarli/Shaarli/pull/2080) but closed their PR before it could be reviewed.

Both methods are supported in the sense that they are documented, and should work out-of-the-box (if they don't, we haven't received any bug report stating so).

If you want to try the updated docker-compose from the PR above and report back, we could merge the updated version.

**Reply by daks** [reply]
> Both methods are supported in the sense that they are documented, and should work out-of-the-box (if they don't, we haven't received any bug report stating so).

I consider security also an important aspect :)

> If you want to try the updated docker-compose from the PR above and report back, we could merge the updated version.

I will look at both and see if I can help for the compose file.

Thanks

**Reply by nodiscc** [reply]
> I consider security also an important aspect :)

As do I :) Hence I opted for other deployment methods (release zip + Debian stable + unattended-upgrades), since Docker images need constant caretaking, manual upgrades and testing.

> I will look at both and see if I can help for the compose file.

Confirmation that these PRs do not break the application would be welcome.

> I plan to use Docker rootless

See also https://github.com/shaarli/Shaarli/pull/1882 (rootless podman)

**Reply by daks** [reply]
> > I consider security also an important aspect :)
> 
> As do I :) Hence I opted for other deployment methods (release zip + Debian stable + unattended-upgrades), since Docker images need constant caretaking, manual upgrades and testing.

Is there any CI/CD enabled for the project? It could be a good way to keep the Docker image up to date

_(I also use Debian+unattended-upgrades but applications still needs to be updated... and I have less time than before to maintain them, I'm looking for ways to reduce this maintenance burden and Docker rootless seems a good way)_

> > I will look at both and see if I can help for the compose file.
> 
> Confirmation that these PRs do not break the application would be welcome.
> 
> > I plan to use Docker rootless
> 
> See also #1882 (rootless podman)

**Reply by nodiscc** [reply]
> Is there any CI/CD enabled for the project?

There is: https://github.com/shaarli/Shaarli/actions/workflows/docker-latest.yml
But the base image version still has to be bumped manually https://github.com/shaarli/Shaarli/pull/2088 and tested.

> applications still needs to be updated... and I have less time than before to maintain them

In the case of Shaarli (and many other applications), releases happen rarely enough that this is not a problem for me (last time I had to bump the version in my ansible role was November 2023 and March 2023 https://github.com/nodiscc/xsrv/commits/master/roles/shaarli/defaults/main.yml). It also gives me a chance to (quickly) read the release notes which should be done in any case
